### PR TITLE
update contrib guide for git-lfs

### LINKED ISSUE #388: Tests: OSError: [Errno 24] Too many open files
There appears to be a leaked file handle somewhere in the tests

- cloned repo
- `pip install -e ".[notebook,docs,tests]"`
- ran `pytest examples`

```
OSError: [Errno 24] Too many open files: '/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pygments/formatters/terminal.py'
```

<details>

```
module = PosixPath('/Users/talley/dev/forks/fastplotlib/examples/desktop/gridplot/gridplot.py'), force_offscreen = None

Traceback (most recent call last):
  File "/Users/talley/miniforge3/envs/fastplotlib/bin/pytest", line 8, in <module>
    sys.exit(console_main())
             ^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/config/__init__.py", line 192, in console_main
    code = main()
           ^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/config/__init__.py", line 169, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/main.py", line 318, in pytest_cmdline_main
    return wrap_session(config, _main)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/main.py", line 306, in wrap_session
    config.hook.pytest_sessionfinish(
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_callers.py", line 130, in _multicall
    teardown[0].send(outcome)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/terminal.py", line 867, in pytest_sessionfinish
    self.config.hook.pytest_terminal_summary(
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pluggy/_callers.py", line 62, in _multicall
    next(wrapper_gen)  # first yield
    ^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/terminal.py", line 882, in pytest_terminal_summary
    self.summary_failures()
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/terminal.py", line 1064, in summary_failures
    self._outrep_summary(rep)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/terminal.py", line 1083, in _outrep_summary
    rep.toterminal(self._tw)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/reports.py", line 89, in toterminal
    longrepr_terminal.toterminal(out)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_code/code.py", line 1072, in toterminal
    element[0].toterminal(tw)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_code/code.py", line 1102, in toterminal
    entry.toterminal(tw)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_code/code.py", line 1197, in toterminal
    self._write_entry_lines(tw)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_code/code.py", line 1179, in _write_entry_lines
    tw._write_source(source_lines, indents)
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_io/terminalwriter.py", line 192, in _write_source
    new_lines = self._highlight(source).splitlines()
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/_pytest/_io/terminalwriter.py", line 203, in _highlight
    from pygments.formatters.terminal import TerminalFormatter
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 936, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1073, in get_code
  File "<frozen importlib._bootstrap_external>", line 1130, in get_data
OSError: [Errno 24] Too many open files: '/Users/talley/miniforge3/envs/fastplotlib/lib/python3.11/site-packages/pygments/formatters/terminal.py'
```

</details>

I'm on macos 14, python 3.11

### PR DESCRIPTION
**Pull request by tlambert03**
closes #388 

@kushalkolar, how's this read?

also, do you ever commit stuff in `diffs/`?  or is it worth adding it to git ignore?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,11 +18,14 @@ cd fastplotlib
...
 
+> If you cloned before installing `git-lfs`, you can run `git lfs pull` at any
+> time to properly download files.
+
 3. Checkout the `main` branch, and then checkout your feature or bug fix branch, and run tests:
 
 > **Warning**
-> Do not commit or add any changes from `examples/screenshots` or `examples/diffs`. 
-> If you are creating new test examples that generate or change screenshots please post an issue on the repo and we will help you. The screenshots will be generated on github actions servers, which you can then copy into the screenshots dir. :) 
+> Do not commit or add any changes from `examples/desktop/screenshots`.
```
Comment (kushalkolar) [inline CONTRIBUTING.md:27]: Keep the diffs part here, if tests produce diffs they will get placed there and these files shouldn't be pushed.

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (tlambert03) [reply CONTRIBUTING.md:27]: I added diffs to git ignore... hence the question in the original post

**Comment by kushalkolar** [comment]
Nevermind you're right sorry! We don't have an empty diffs dir in main (@clewis7 wrote the entire test suite so I'm sometimes unfamiliar with the details here)

**Comment by tlambert03** [comment]
Right, so the main question (for you all), is whether you ever intend to check stuff into diffs, or if it's always the byproduct product of a test.  If the latter, just add it to gitignore, and then developers don't need to pay any special attention to it.

**Comment by kushalkolar** [comment]
Yup it's indeed the latter!

**Comment by kushalkolar** [comment]
Bypassing CI for this since no source files are touched.

**Pull request merged**